finality-aleph/src/lib.rs

```diff
@@ -270,4 +270,5 @@ pub struct AlephConfig<C, SC> {
     pub rate_limiter_config: RateLimiterConfig,
     pub sync_oracle: SyncOracle,
     pub validator_address_cache: Option<ValidatorAddressCache>,
+    pub transaction_pool_info_provider: T,
```

Comment:
I also see an option of using the concrete type here. The whole project (talking bout `AlephConfig` and `run_validator_node`) doesn't need to be parameterized by transaction pool info provider. You only need the abstraction for unit tests of `metrics::chain_state`.

Reply:
Agree, now I pass `TransactionPool` from substrate and don't export internal wrapper (which still has to be used, because we have to store import stream somewhere).

Reply:
I'm happy with this approach, so @woocash2 please resolve this if you are as well or comment if you aren't.

Reply:
I haven't seen a way to not parametrize `AlephConfig` with tx pool (other than using box) - unfortunately, unit (or rather integration) tests have to be in finality-aleph and use `BasicPool` as well.

Reply:
Looks aight to me. If up to me, it's a resolve, I'll let @timorleph to take a look and resolve, since at the end the solution aligns to his initial proposal.